microsoft_auth/context_processors.py

```diff
@@ -41,10 +41,15 @@ def microsoft(request):
                     "for your `redirect_uri` is `localhost`\n"
                 )
 
-    # initialize Microsoft client using CSRF token as state variable
-    signer = TimestampSigner()
-    state = signer.sign(get_token(request))
-    microsoft = MicrosoftClient(state=state, request=request)
+    # Initialize Microsoft client using dict with CSRF token and optional
+    # next path as state variable
+    next_ = request.GET.get("next")
+    state = dict(token=get_token(request))
+    if next_:
+        state["next"] = next_
+
+    signed_state = dumps(state, salt="microsoft_auth")
```

Comment:
Very nice. I saw the email notification for this and I was like "oh no, here we go again", but signing the redirect is a great way to handle this.